app/frontend/src/features/communities/leaveReference/LeaveReferencePage.test.tsx

```diff
@@ -0,0 +1,187 @@
+    getAvailableReferencesMock.mockImplementation(getAvailableReferences);
+  });
+
+  describe("When the reference type is invalid", () => {
+    beforeEach(() => {
+      renderLeaveFriendReferencePage("hello", "1");
+    });
+
+    it("Returns an error", async () => {
+      const errorAlert = await screen.findByRole("alert");
+      expect(
+        within(errorAlert).getByText(INVALID_REFERENCE_TYPE)
+      ).toBeVisible();
+    });
+
+    it("does not show the form", () => {
+      expect(
+        screen.queryByRole("heading", {
+          name: "You met with Friendly Cow",
+        })
+      ).not.toBeInTheDocument();
+    });
+  });
+
+  describe("When the reference type is friend", () => {
+    describe("and the users are friends", () => {
+      beforeEach(() => {
+        renderLeaveFriendReferencePage("friend", "5");
+      });
+
+      it("verifies that the review type is available", async () => {
+        expect(getAvailableReferencesMock).toHaveBeenCalledTimes(1);
+        expect(getAvailableReferencesMock).toHaveReturned();
+      });
+
+      it("does not return an error", () => {
+        expect(screen.queryByRole("alert")).not.toBeInTheDocument();
+      });
+
+      it("displays the form", async () => {
+        expect(
+          await screen.findByRole("heading", {
+            name: "You met with Friendly Cow",
+          })
+        ).toBeInTheDocument();
+      });
+    });
+
+    describe("and the users aren't friends", () => {
+      beforeEach(() => {
+        renderLeaveFriendReferencePage("friend", "1");
+      });
+
+      it("verifies the review type", async () => {
+        expect(getAvailableReferencesMock).toHaveBeenCalledTimes(1);
+        expect(getAvailableReferencesMock).toHaveReturned();
+      });
+
+      it("Returns an error", async () => {
+        const errorAlert = await screen.findByRole("alert");
+        expect(
+          within(errorAlert).getByText(REFERENCE_TYPE_NOT_AVAILABLE)
+        ).toBeVisible();
+      });
+
+      it("does not show the form", () => {
+        expect(
+          screen.queryByRole("heading", {
+            name: "You met with Funny Cat current User",
+          })
+        ).not.toBeInTheDocument();
+      });
+    });
+  });
+
+  describe("When the reference type is hosted/surfed", () => {
+    describe("And a review is available", () => {
+      beforeEach(() => {
+        renderLeaveRequestReferencePage("hosted", "5", 1);
+      });
+
+      it("verifies that the review type is available", async () => {
+        expect(getAvailableReferencesMock).toHaveBeenCalledTimes(1);
+        expect(getAvailableReferencesMock).toHaveReturned();
+      });
+
+      it("does not return an error", () => {
+        expect(screen.queryByRole("alert")).not.toBeInTheDocument();
+      });
+
+      it("displays the form", async () => {
+        expect(
+          await screen.findByRole("heading", {
+            name: "You met with Friendly Cow",
```

Comment:
Hmm I think it'd be good if we display a slightly different title for host/surfing references like "You surfed with Friendly Cow" or something to distinguish between them